src/python/pants/backend/project_info/count_loc.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Tuple
+
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalTool,
+    ExternalToolRequest,
+)
+from pants.engine.console import Console
+from pants.engine.fs import Digest, MergeDigests, SourcesSnapshot
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.platform import Platform
+from pants.engine.process import Process, ProcessResult
+from pants.engine.rules import Get, collect_rules, goal_rule
+from pants.option.custom_types import shell_str
+from pants.util.enums import match
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+
+class SuccinctCodeCounter(ExternalTool):
+    """The Succinct Code Counter (https://github.com/boyter/scc)."""
+
+    options_scope = "succinct-code-counter"
+    default_version = "2.12.0"
+    default_known_versions = [
+        "2.12.0|darwin|70b7002cd1e4541cb37b7b9cbc0eeedd13ceacb49628e82ab46332bb2e65a5a6|1842530",
+        "2.12.0|linux|8eca3e98fe8a78d417d3779a51724515ac4459760d3ec256295f80954a0da044|1753059",
+    ]
+
+    def generate_url(self, plat: Platform) -> str:
+        plat_str = match(plat, {Platform.darwin: "apple-darwin", Platform.linux: "unknown-linux"})
+        return (
+            f"https://github.com/boyter/scc/releases/download/v{self.version}/scc-{self.version}-"
+            f"x86_64-{plat_str}.zip"
+        )
+
+    def generate_exe(self, _: Platform) -> str:
+        return "./scc"
+
+
+class CountLinesOfCodeSubsystem(GoalSubsystem):
+    """Count lines of code using Succinct Code Counter (https://github.com/boyter/scc)."""
+
+    name = "count-loc"
+
+    @classmethod
+    def register_options(cls, register) -> None:
```

Comment:
Should this be on the tool subsystem? `--scc-args` is more ergonomic. And in a future where (for whatever reason) we support more than one code counter, this would keep things nicely compartmentalized.

Reply:
True. I forgot that we could still use passthrough args style, which I think we do want to use.